test/app-for-e2e/pkg/handlers/handlers.go

```diff
@@ -112,16 +91,117 @@ func NewMemLoadHandler(res http.ResponseWriter, req *http.Request) {
-	log.Println("Entered ListRoutesHandler")
-	paths := []string{"cpu", "mem"}
-	for _, path := range paths {
-		fmt.Fprintln(res, path)
+	paths := "paths: / , cpu , newmem , pet , pupulate"
+	messageResponseJSON(res, http.StatusOK, paths)
+	return
+}
+
+// DEMO APP
+
+// MessageResponse contains any additional information required in the server response
+type MessageResponse struct {
+	Message string `json:"message"`
+}
+
+// PetHandler handles pets and pet accessories
+// ex: ex: localhost:1738/pet , ex: localhost:1738/pet?petId=4
+func PetHandler(res http.ResponseWriter, req *http.Request) {
+	petIdParam, _ := req.URL.Query()["petId"]
+
+	//GET
+	if req.Method == http.MethodGet {
+		if len(petIdParam) < 1 {
+			message := "Pet ID required"
+			messageResponseJSON(res, http.StatusBadRequest, message)
+			return
+		}
+		petId, err := strconv.Atoi(petIdParam[0])
+		if err != nil {
+			message := "Could not parse Pet ID; Ensure ID consists of numerics only"
+			messageResponseJSON(res, http.StatusBadRequest, message)
+			return
+		}
+		pet, err := db.GetPetByID(petId)
+		if err != nil {
+			message := "Pet not found: " + err.Error()
+			messageResponseJSON(res, http.StatusNotFound, message)
+			return
+		}
+		jsonResponse(res, http.StatusOK, pet)
+		return
+	}
+
+	//POST
+	if req.Method == http.MethodPost {
+		body, err := ioutil.ReadAll(req.Body)
+		defer req.Body.Close()
+		if err != nil {
+			message := "Invalid request; body required"
+			messageResponseJSON(res, http.StatusMethodNotAllowed, message)
+			return
+		}
+
+		var pet db.Pet
+		err = json.Unmarshal(body, &pet)
+		if err != nil {
+			fmt.Println(err)
+			message := "Invalid request; can't unmarshal"
+			messageResponseJSON(res, http.StatusMethodNotAllowed, message)
+			return
+		}
+
+		err = db.AddPet(pet)
+		if err != nil {
+			message := "Invalid request; can't add pet to DB"
+			messageResponseJSON(res, http.StatusMethodNotAllowed, message)
+			return
+		}
+		jsonResponse(res, http.StatusOK, "Pet Added")
+		return
+	}
+
+	//DELETE
+	if req.Method == http.MethodDelete {
+		if len(petIdParam) < 1 {
+			message := "Pet ID required"
+			messageResponseJSON(res, http.StatusBadRequest, message)
+			return
+		}
+		petId, err := strconv.Atoi(petIdParam[0])
+		if err != nil {
+			message := "Could not parse Pet ID; Ensure ID consists of numerics only"
+			messageResponseJSON(res, http.StatusBadRequest, message)
+			return
+		}
+		err = db.DeletePet(petId)
+		if err != nil {
+			message := "Could remove pet from DB"
+			messageResponseJSON(res, http.StatusNotFound, message)
+			return
+		}
+		jsonResponse(res, http.StatusOK, "Pet Deleted")
+		return
 	}
-	log.Println("Leaving ListRoutesHandler")
 }
 
+// PupulateHandler populates the Pets table with pups
+// ex: localhost:1738/pupulate?num=1000
+func PupulateHandler(res http.ResponseWriter, req *http.Request) {
+	numPups := parseOrDefault(req, "num", 100)
+	err := db.PopulateTable(numPups)
```

Comment:
I should prob make `db.PopulateTable` **Pupulate** instead so the exposed endpoint is more clear

Reply:
up to you on that one. You could make the argument the db layer is separated from the pups